app/scripts/modules/core/src/pipeline/config/stages/managed/importDeliveryConfigStage.ts

```diff
@@ -0,0 +1,30 @@
+import { Registry } from 'core/registry';
+import { ExecutionDetailsTasks } from 'core/pipeline';
+
+import { ImportDeliveryConfigStageConfig } from './ImportDeliveryConfigStageConfig';
+import { ImportDeliveryConfigExecutionDetails } from './ImportDeliveryConfigExecutionDetails';
+import { IStageOrTriggerBeforeTypeValidationConfig } from 'core/pipeline/config/validation/stageOrTriggerBeforeType.validator';
+import { SETTINGS } from 'core/config';
+
+if (SETTINGS.feature.managedDelivery) {
+  Registry.pipeline.registerStage({
+    label: 'Import Delivery Config',
+    description:
+      "Retrieve a Delivery Config manifest from the git repository configured in the pipeline's trigger, then update it in Spinnaker.",
+    extendedDescription: `<a target="_blank" href="https://www.spinnaker.io/reference/managed-delivery/gitops.md">
+      <span class="small glyphicon glyphicon-file"></span> Documentation</a>`,
+    key: 'importDeliveryConfig',
+    restartable: false,
+    component: ImportDeliveryConfigStageConfig,
+    executionDetailsSections: [ImportDeliveryConfigExecutionDetails, ExecutionDetailsTasks],
+    //validateFn: validate,
+    validators: [
+      {
+        type: 'stageOrTriggerBeforeType',
+        // FIXME: this needs to be generalized so that it accepts a rocket trigger in the netflix deployment
+        stageType: 'git',
+        message: 'This stage requires a git trigger to locate your Delivery Config manifest.',
```

Comment:
```suggestion
        getMessage: labels => `
          This stage requires a stage or trigger that provides repository information to locate your Delivery Config manifest:
          <ul>${labels.map(label => `<li>${label}</li>`)}</ul>
          `
```
not sure how to make

Reply:
@anotherchrisberry Not sure what you were going to finish that sentence with, but if I still need to declare the types of triggers that provide the info I want, this won't work because I can't introduce Netflix-specific trigger names in `stageTypes` here... I might end up just having to write a validation function that checks if the required properties are available in the trigger.

Reply:
Oh, sorry, you don't need to declare the trigger types - you just need to update our internal trigger and add `providesRepositoryInformation: true` to its definition, i.e.
```javascript
Registry.pipeline.registerTrigger({
  label: 'our internal trigger type',
  // ...
  providesRepositoryInformation: true, // add this
  providesVersionForBake: true,
  // ...
});
```

Reply:
Oh, that makes much more sense! I'll undo my change that introduced a validate function.